pkg/policy/mapstate.go

```diff
@@ -774,9 +893,14 @@ func (ms *mapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapState
 	allCpy := allKey
 	allCpy.TrafficDirection = newKey.TrafficDirection
 	// If we have a deny "all" we don't accept any kind of map entry.
-	if _, ok := ms.denies[allCpy]; ok {
+	if _, ok := ms.denies.Lookup(allCpy); ok {
 		return
 	}
+
+	// We cannot update the map while we are
+	// iterating through it, so we record the
+	// changes to be made and then apply them.
+	var updates []MapChange
```

Comment:
nit: The change here makes sense, wondering about whether it's worth optimizing memory allocations here; we often preallocate slices to expected capacities to avoid reallocating when resizing them. I guess if this is for a single insert, there should be a pretty clear maximum bound on the number of MapChanges to accumulate based on the patterns for the insert?

Reply:
I can't think of a good heuristic for knowing this ahead of time. The most common outcome is probably no updates at all.